### ISSUE TITLE
`expand` throws an error for certain factorised expressions

### ISSUE THREAD

**Issue by augustinas1** [JuliaSymbolics/SymbolicUtils.jl#305]
This seems to be a new bug (only in v0.13). Here's a minimal example:
```julia
@syms a b
expand(2*a*(a+b))
````
```julia
ERROR: MethodError: no method matching polynomialtype(::Type{Any}, ::Type{Int64})
Closest candidates are:
  polynomialtype(::Union{AbstractVector{PT}, Type{var"#s9"} where var"#s9"<:AbstractVector{PT}}, ::Type{T}) where {PT<:(MultivariatePolynomials.AbstractPolynomialLike), T} at C:\Users\asukys\.julia\packages\MultivariatePolynomials\knQW3\src\polynomial.jl:102
  polynomialtype(::Union{DynamicPolynomials.Monomial{C}, DynamicPolynomials.PolyVar{C}, Type{var"#s20"} where var"#s20"<:Union{DynamicPolynomials.Monomial{C}, DynamicPolynomials.PolyVar{C}, DynamicPolynomials.Polynomial{C, T} where T, DynamicPolynomials.Term{C, T} where T}, DynamicPolynomials.Polynomial{C, T} where T, DynamicPolynomials.Term{C, T} where T}, ::Type{T}) where {C, T} at C:\Users\asukys\.julia\packages\DynamicPolynomials\5p8Ga\src\DynamicPolynomials.jl:41
  polynomialtype(::Union{Type{M}, M}, ::Type{T}) where {M<:MultivariatePolynomials.AbstractMonomialLike, T} at C:\Users\asukys\.julia\packages\MultivariatePolynomials\knQW3\src\polynomial.jl:99   
  ...
Stacktrace:
  [1] polynomialtype(#unused#::Type{MultivariatePolynomials.AbstractPolynomialLike{Int64}}, #unused#::Type{Int64})
    @ MultivariatePolynomials C:\Users\asukys\.julia\packages\MultivariatePolynomials\knQW3\src\polynomial.jl:100
  [2] promote_rule_constant(#unused#::Type{Int64}, PT::Type{MultivariatePolynomials.AbstractPolynomialLike{Int64}})
    @ MultivariatePolynomials C:\Users\asukys\.julia\packages\MultivariatePolynomials\knQW3\src\promote.jl:27
  [3] promote_rule(#unused#::Type{Int64}, #unused#::Type{MultivariatePolynomials.AbstractPolynomialLike{Int64}})
    @ MultivariatePolynomials C:\Users\asukys\.julia\packages\MultivariatePolynomials\knQW3\src\promote.jl:30
  [4] promote_type
    @ .\promotion.jl:233 [inlined]
  [5] promote_eltypeof(v1::Int64, vs::Vector{MultivariatePolynomials.AbstractPolynomialLike{Int64}})
    @ Base .\abstractarray.jl:1460
  [6] _cat(::Val{1}, ::Int64, ::Vararg{Any, N} where N)
    @ Base .\abstractarray.jl:1641
  [7] #cat#129
    @ .\abstractarray.jl:1781 [inlined]
  [8] vcat
    @ .\abstractarray.jl:1710 [inlined]
  [9] arguments(a::SymbolicUtils.Mul{Number, Int64, Dict{Any, Number}, Nothing})
    @ SymbolicUtils C:\Users\asukys\.julia\packages\SymbolicUtils\BKm0J\src\types.jl:773
 [10] (::SymbolicUtils.Rewriters.Walk{:post, SymbolicUtils.Rewriters.PassThrough{SymbolicUtils.Rewriters.RestartedChain{Vector{SymbolicUtils.AbstractRule}}}, SymbolicUtils.var"#simterm#125"{SymbolicUtils.var"#simterm#114#126"}, false})(x::SymbolicUtils.Mul{Number, Int64, Dict{Any, Number}, Nothing})
    @ SymbolicUtils.Rewriters C:\Users\asukys\.julia\packages\SymbolicUtils\BKm0J\src\rewriters.jl:142
 [11] (::SymbolicUtils.Rewriters.Fixpoint{SymbolicUtils.Rewriters.Walk{:post, SymbolicUtils.Rewriters.PassThrough{SymbolicUtils.Rewriters.RestartedChain{Vector{SymbolicUtils.AbstractRule}}}, SymbolicUtils.var"#simterm#125"{SymbolicUtils.var"#simterm#114#126"}, false}})(x::SymbolicUtils.Mul{Number, Int64, Dict{Any, Number}, Nothing})
    @ SymbolicUtils.Rewriters C:\Users\asukys\.julia\packages\SymbolicUtils\BKm0J\src\rewriters.jl:104
 [12] to_mpoly(t::SymbolicUtils.Mul{Number, Int64, Dict{Any, Number}, Nothing}, variable_type::Type, dicts::Tuple{OrderedCollections.OrderedDict{SymbolicUtils.Sym, Any}, OrderedCollections.OrderedDict{Any, SymbolicUtils.Sym}})
    @ SymbolicUtils C:\Users\asukys\.julia\packages\SymbolicUtils\BKm0J\src\abstractalgebra.jl:91
 [13] to_mpoly
    @ C:\Users\asukys\.julia\packages\SymbolicUtils\BKm0J\src\abstractalgebra.jl:78 [inlined]
 [14] expand(expr::SymbolicUtils.Mul{Number, Int64, Dict{Any, Number}, Nothing}, variable_type::Type) (repeats 2 times)
    @ SymbolicUtils C:\Users\asukys\.julia\packages\SymbolicUtils\BKm0J\src\abstractalgebra.jl:173
 [15] top-level scope
    @ none:1
```

**Comment by blegat** [comment]
Should have been fixed by https://github.com/JuliaAlgebra/MultivariatePolynomials.jl/pull/166 which is included in v0.3.18 of MultivariatePolynomials

**Comment by augustinas1** [comment]
Yep it works fine now, thanks @blegat!
